cmd/kueue/main.go

```diff
@@ -194,7 +194,7 @@ func setupIndexes(ctx context.Context, mgr ctrl.Manager, cfg *configapi.Configur
 	if features.Enabled(features.ProvisioningACC) {
 		if !provisioning.ServerSupportsProvisioningRequest(mgr) {
 			setupLog.Error(nil, "Provisioning Requests are not supported, skipped admission check controller setup")
-		} else if err := provisioning.SetupIndexer(ctx, mgr.GetFieldIndexer()); err != nil {
+		} else if err := provisioning.SetupIndexer(ctx, mgr.GetFieldIndexer(), mgr.GetScheme()); err != nil {
```

Comment:
This change might not be necessary if we pass the GVK instead of the empty `ProvisioningRequestConfig` object?

Reply:
but `SetupIndexer` anyway has the type `ProvisioningRequestConfig` hardcoded. Going via creating the object based on type, then getting back type from the object seems unnecessary. 

We do something similar for the various job integrations, for example here we memorize the gvk for MPIJob and use in `SetupIndexer` function: https://github.com/kubernetes-sigs/kueue/blob/a2fffa4686c47fada07d8a02566408566c5e163b/pkg/controller/jobs/mpijob/mpijob_controller.go#L201C60-L201C63/

Reply:
done